Assets/Features/Managers/BuildingManager.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Features.Building.Scripts.Datatypes;
+using Features.EventManager;
+using UnityEngine;
+
+namespace Features.Managers
+{
+    public class BuildingManager : MonoBehaviour
+    {
+        [field: SerializeField] public BuildableObject CurrentBuildableObject { get; private set; }
```

Comment:
Is the `field` part in the tag necessary? It looks weird

Reply:
Field tag is necessary when using a private set for a property when you want it editable in the inspector